static/js/containers/pages/DashboardPage.js

```diff
@@ -77,13 +77,20 @@ export class DashboardPage extends React.Component<
     submittingEnrollmentId:       null,
     activeMenuIds:                [],
     activeEnrollMsgIds:           [],
-    emailSettingsModalVisibility: false
+    emailSettingsModalVisibility: []
   }
 
-  toggleEmailSettingsModalVisibility = () => {
+  toggleEmailSettingsModalVisibility = (enrollmentId: number) => {
     const { emailSettingsModalVisibility } = this.state
+    let isOpen = false
+    if (emailSettingsModalVisibility[enrollmentId] === undefined) {
+      isOpen = true
+    } else {
+      isOpen = !emailSettingsModalVisibility[enrollmentId]
+    }
```

Comment:
I see this same logic for checking which items being opened have been used on multiple places can you convert that into a method and call it here and on the other place?

Reply:
It's not the same because Line#89 is updating the value

Reply:
Line#87 setting `isOpen=true` but in the next occurrence it's consider as false